src/screens/minigames/base.py

```diff
@@ -0,0 +1,71 @@
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import TypeVar
+
+import pygame
+
+
+class MinigameSetupError(Exception):
+    pass
```

Comment:
Thanks!
I had already changed this locally, but wanted to wait with pushing it until I was done adding a minigame cutscene. However, since I decided to zoom the camera out during the cutscene to show the entire minigame area, this involves a few more changes that I'm still working on, and will probably fit better in a separate minor pull request 😄